Add refresh content docs (#26106)
gatsbyjs/gatsby: docs/docs/refreshing-content.md

---
title: Refreshing Content
---

> During local development it can be useful to refresh sourced content without restarting the development server. To faciliate this Gatsby exposes an environmental variable `ENABLE_GATSBY_REFRESH_ENDPOINT`.

Reviewer: ```suggestion
During local development, it can be useful to refresh sourced content without restarting the development server. To facilitate this, Gatsby exposes an environment variable called `ENABLE_GATSBY_REFRESH_ENDPOINT`.
```

> If set to true, this will expose a /\_\_refresh webhook that is able to receive POST requests to refresh the sourced content. This exposed webhook can be triggered whenever remote data changes.

Reviewer: ```suggestion
If set to `true`, this will expose a `/__refresh` webhook that can receive POST requests to refresh the sourced content. This exposed webhook can be triggered whenever remote data changes.
```

> You can trigger this endpoint locally, for example, on Unix-based operating systems (like Ubuntu and macOS) using curl -X POST http://localhost:8000/__refresh.

Reviewer: ```suggestion
You can trigger this endpoint locally, for example, on Unix-based operating systems (like Ubuntu and macOS) using `curl -X POST http://localhost:8000/__refresh`.
```

> Additionally, the sourced content can also be refreshed through the GraphiQL explorer.

Reviewer: ```suggestion
Additionally, the sourced content can also be refreshed through the [GraphiQL explorer](/docs/running-queries-with-graphiql).
```

This is more of a suggestion than correction! I think adding this link might be helpful but I'm happy to leave it to your discretion.

Reviewer: In addition, it's not clear from this sentence that there's now a button people can click in order to refresh. Could you get a little more specific here? Maybe something like...

```suggestion
Additionally, the sourced content can also be refreshed with the "Refresh Data" button in the [GraphiQL explorer](/docs/running-queries-with-graphiql). This button is only visible if `ENABLE_GATSBY_REFRESH_ENDPOINT` is set to `true`.
```

> Additionally, the sourced content can also be refreshed with the "Refresh Data" button in the [GraphiQL explorer](/docs/running-queries-with-graphiql). This button is only visible if `ENABLE_GATSBY_REFRESH_ENDPOINT` is set to `true`.

Reviewer 2: should we add a backlink to this doc at the GraphiQL Exlorer page?
